libraries/commerce/product/selectors/product.js

```diff
@@ -531,7 +541,9 @@ export const isBaseProduct = createSelector(
       return null;
     }
 
-    return productData.baseProductId === null;
+    const hasVariants = productData.flags && productData.flags.hasVariants;
```

Comment:
Perhaps a comment would help others to understand why these checks are necessary.

Reply:
I've tried to add a suitable description.